### DISCUSSION TITLE
Tests timing out with this error: CertVerifyProcBuiltin

### DISCUSSION THREAD

**Discussion by jazanne** [cypress-io/cypress#22128] [Questions and Help] [answered]
A few weeks ago these errors starting popping up while running tests. We run the tests via GitHub Actions and have a timeout set on the job step, which is how they eventually "fail".

Using Cypress 9.7.0 & Cypress GitHub Action version 3.1.0.

We have been running these tests regularly for months with no issues and no significant changes to the tests, other than upgrading versions of cypress.

Any ideas on recent changes that may have prompted this?

```
[3801:0606/152837.383892:ERROR:cert_verify_proc_builtin.cc(681)] CertVerifyProcBuiltin for www.googletagmanager.com failed:
----- Certificate i=0 (OU=Cypress Proxy Server Certificate,O=Cypress Proxy CA,L=Internet,ST=Internet,C=Internet,CN=www.googletagmanager.com) -----
ERROR: No matching issuer found
[3801:0606/152837.390166:ERROR:cert_verify_proc_builtin.cc(681)] CertVerifyProcBuiltin for cdn.segment.io failed:
----- Certificate i=0 (OU=Cypress Proxy Server Certificate,O=Cypress Proxy CA,L=Internet,ST=Internet,C=Internet,CN=cdn.segment.io) -----
ERROR: No matching issuer found
[3801:0606/152837.847154:ERROR:cert_verify_proc_builtin.cc(681)] CertVerifyProcBuiltin for www.google-analytics.com failed:
----- Certificate i=0 (OU=Cypress Proxy Server Certificate,O=Cypress Proxy CA,L=Internet,ST=Internet,C=Internet,CN=www.google-analytics.com) -----
ERROR: No matching issuer found
[3801:0606/152838.158565:ERROR:cert_verify_proc_builtin.cc(681)] CertVerifyProcBuiltin for rum.browser-intake-datadoghq.com failed:
----- Certificate i=0 (OU=Cypress Proxy Server Certificate,O=Cypress Proxy CA,L=Internet,ST=Internet,C=Internet,CN=rum.browser-intake-datadoghq.com) -----
ERROR: No matching issuer found
```

**Comment by dswindle** [comment]
We're seeing the same behavior. Also running Cypress 9.7.0. We are running via Jenkins (not via Github actions).

**Comment by olemarius** [comment]
Getting the same here after upgrading to 10, seems tests are randomly failing.

**Comment by brendaneross** [comment]
I'm also seeing this error running Cypress 9.7.0 in Jenkins. A high number of tests are being passed over, and I'm tempted to rebuild the pipeline from scratch at this point.

**Comment by just-toby** [comment]
i'm also seeing this error using cypress 10.1.0 and 10.0.1 

this is with Github actions, trying with all the different browsers. the error seems to block network requests to all external servers. we don't want to mock the data because we'd prefer to test it fully e2e

**Reply by Diljotv3** [reply]
@just-toby did u find the solution for it since i am having the same problem for a while

**Comment by shirblc** [comment]
Seeing the same with Cypress 9.7.0 running in Google's Cloudbuild... anyone has any solutions/workarounds?

**Comment by KyorCode** [comment]
Same here Cypress 10.1.0 running in Azure pipelines

**Comment by oegea** [comment]
We're getting the same error with Cypress 10.1.0, wrapped in a Docker image and executed by Travis.

**Comment by davidfrigolet** [comment]
same error with Cypress 10.1.0, wrapped in a Docker image and executed by CircleCI

**Comment by revolalex** [comment]
idem, same error with Cypress 10.1.0, wrapped in a Docker image and executed by CircleCI

**Comment by KyorCode** [comment]
After some testing, it seems to me that this is an issue in electron. setting the browser for headless runs to chrome solved this for me.

`cypress run --headless --browser chrome`

Still requires attention for electron though!

**Reply by adirksen** [reply]
This solved the certificate errors being thrown but we are still seeing all external network requests being blocked.

**Reply by oegea** [reply]
We are not seeing warnings after running our tests on Chrome in headless mode.

**Reply by FeigyHaim** [reply]
> This solved the certificate errors being thrown but we are still seeing all external network requests being blocked.

@adirksen I'm also seeing network calls blocked , causing tests to fail randomly, using electron, and also getting the errors above? did you manage to fix your issue? how?

**Comment by just-toby** [comment]
looks like there was [an issue ](https://github.com/cypress-io/cypress/issues/22196) already closed because the team needs an isolated reproducible example to solve this.

~~fwiw, using ` cypress-io/github-action@v4` with browser=chrome and headless did _not_ solve the issue for us. again, we're using GH actions. not sure if it matters, but we're trying to test a next.js app~~

edit: headless mode actually did solve this issue for my team

**Comment by liambutler** [comment]
We're also seeing this in our electron runs, and would rather not switch to Chrome headless if it can be avoided. These error messages are generating a lot of noise and causing confusion when diagnosing failure causes

**Comment by flotwig** [comment, marked as answer]
Hi folks, I can confirm that this is not what is causing your tests to time out - this "error" should not be appearing, but it is not affecting the run of Cypress or your tests' ability to make HTTPS requests, it is just an internal debug log of Chromium.

#22342 just merged and will suppress this stderr, it will land in Cypress 10.2.0.

Other than that, I would search the issue tracker for other issues related to hanging tests in CI - there are a few.

**Reply by MichaelDimmitt** [reply]
can confirm adding ^10.2.0 to package.json got rid of warning on my end.
and my tests are passing.

**Comment by adirksen** [comment]
Hi, we were able to figure out that the issue was being caused by a misconfiguration between our environment variables on the GitHub actions runner and our doppler secrets integration.
Right now we are running in headless chrome and that seems to have suppressed the logs.

Thanks for your help and responsiveness!

**Comment by andrii-lundiak** [comment]
Jan-2023, I upgraded to Cypress `9.7.0` and I started to have this issue.
I have a few *.spec.ts` files, and as soon as spec finishes I see on my CI logs this output:
![213871423-0bba3eaf-32a2-4b11-8242-cc62262acf90](https://user-images.githubusercontent.com/85486996/213871748-82ae2800-0e7a-46ba-a578-50633ffbc4c2.png)


Note. YES, if it's reasoab==nale to mention, but my main application page has connections to: 
- My main host
- `dc.services.visualstudio.com` (Azure Application Insights), 
- `www.google-analytics.com` (Google Analytics)
- `sentry.io` (Sentry)

Also, my tests do `cy.visit()` a few times during executions, so Electron/Chrome pages has been reloaded.

I'm going to upgrade to latest Cypress `v10` but I may be delayed, so for now I would like to know what exact v10 version has been really fixed it, coz `10.2.0` and also `10.8.0` (https://github.com/cypress-io/cypress/issues/7307) seems to be not.

**Reply by andrii-lundiak** [reply]
UPD. I tried `cypress run --headless --browser chrome` (as suggested [here](https://stackoverflow.com/questions/72613532/running-cypress-tests-in-docker-image-cypress-base17-3-0-gives-error-certveri))  and it seems to be helpful - I got rid of `ERROR: No matching issuer found` but instead I now see known Chrome issue

[2258:0121/160445.929560:ERROR: 
**sandbox_linux.cc**(377)] InitializeSandbox() called with multiple threads in process **gpu-process**.

[2258:0121/160445.933093:ERROR: 
**gpu_memory_buffer_support_x11.cc**(44)] dri3 extension not supported.

At least now it really hints that that initial is an Electron issue. Cypress [changelog for 9.7.0](https://docs.cypress.io/guides/references/changelog#9-7-0) shows that there were Electron-related changes. But also were a few for almost all 10.x versions of Cypress.
